src/controlCenter/widgets/backlight/backlightUtil.vala

```diff
@@ -0,0 +1,113 @@
+namespace SwayNotificationCenter.Widgets {
+    class BacklightUtil {
+
+        [DBus (name = "org.freedesktop.login1.Session")]
+        interface Login1 : Object {
+            public abstract void set_brightness (string subsystem, string name, uint32 brightness) throws GLib.Error;
+        }
+
+        string path_current;
+        string path_max;
+        File fd;
+        FileMonitor monitor;
+
+        int max;
+
+        Login1 login1;
+        string device;
+        string subsystem;
+
+        public signal void brightness_change (int percent);
+
+        public BacklightUtil (string s, string d) {
+            this.subsystem = s;
+            this.device = d;
+
+            path_current = Path.build_filename ("/sys/class/" + subsystem + "/" + device + "/brightness");
+            path_max = Path.build_filename ("/sys/class/" + subsystem + "/" + device + "/max_brightness");
+            fd = File.new_for_path (path_current);
```

Comment:
Check if file exists with `fd.query_exists ()`. Currently segfaults when the file doesn't exist so don't run `set_max_value` if it doesn't exist. The widget should be hidden/removed if there's no device found